### PR TITLE
Update Tokens::isMonolithicPhp

### PR DESCRIPTION
**Pull request by kubawerlos**

### CODE REVIEW

File: src/Tokenizer/Tokens.php [hunk 1]
```diff
@@ -1088,16 +1088,18 @@ public function clearRange(int $indexStart, int $indexEnd): void
...
             return false;
         }
 
-        if ($this->isTokenKindFound(T_INLINE_HTML)) {
+        if ($this->countTokenKind(T_INLINE_HTML) > 1) {
             return false;
         }
 
+        if (1 === $this->countTokenKind(T_INLINE_HTML)) {
+            return 1 === Preg::match('/^#!.*$/m', $this[0]->getContent());
```
Comment (SpacePossum) [inline src/Tokenizer/Tokens.php:1100]: not 100% sure about the `m` modifier, I was thing about something like:

```
yield [false, "#!/usr/bin/php\nFoo bar<?php\n"];
yield [false, "#!/usr/bin/env php -n \nFoo bar\n<?php\n"];
```

Comment (keradus) [reply src/Tokenizer/Tokens.php:1100]: :+1:
plus the variants with concrete php, eg php7.4, php7.4-cli, php8 etc

File: src/Tokenizer/Tokens.php [hunk 2]
```diff
@@ -1088,16 +1088,18 @@ public function clearRange(int $indexStart, int $indexEnd): void
...
+        }
+
         return 1 >= ($this->countTokenKind(T_OPEN_TAG) + $this->countTokenKind(T_OPEN_TAG_WITH_ECHO));
```
Comment (keradus) [inline src/Tokenizer/Tokens.php:1103]:
```suggestion
        return 1 === ($this->countTokenKind(T_OPEN_TAG) + $this->countTokenKind(T_OPEN_TAG_WITH_ECHO));
```

do we have a case when we may have no T_OPEN_TAG, no T_OPEN_TAG_WITH_ECHO, file is non-empty, and we still claim it's monolithic?

File: tests/Tokenizer/TokensTest.php [hunk 3]
```diff
@@ -353,9 +353,12 @@ public function provideMonolithicPhpDetectionCases(): iterable
     {
         yield [true, "<?php\n"];
         yield [true, "<?php\n?>"];
+        yield [true, "#!/usr/bin/php\n<?php\n"];
```
Comment (keradus) [inline tests/Tokenizer/TokensTest.php:356]:
```suggestion
        yield [true, "#!/usr/bin/php\n<?php\n"];
        yield [true, "#!/usr/bin/php8\n<?php\n"];
```

File: tests/Tokenizer/TokensTest.php [hunk 4]
```diff
@@ -353,9 +353,12 @@ public function provideMonolithicPhpDetectionCases(): iterable
...
+        yield [true, "#!/usr/bin/env php\n<?php\n"];
+        yield [true, "#!/usr/bin/env php7.4\n<?php\n"];
```
Comment (keradus) [inline tests/Tokenizer/TokensTest.php:358]:
```suggestion
        yield [true, "#!/usr/bin/env php7.4\n<?php\n"];
        yield [true, "#!/usr/bin/env php7.4-cli\n<?php\n"];
```

File: tests/Tokenizer/TokensTest.php [hunk 5]
```diff
@@ -353,9 +353,12 @@ public function provideMonolithicPhpDetectionCases(): iterable
...
+        yield [true, "#!/usr/bin/env this-is\ntoo-much\n<?php\n"];
         yield [false, ''];
         yield [false, ' '];
-        yield [false, "#!/usr/bin/env php\n<?php\n"];
         yield [false, " <?php\n"];
         yield [false, "<?php\n?> "];
         yield [false, "<?php\n?><?php\n"];
```
Comment (keradus) [inline tests/Tokenizer/TokensTest.php:364]:
```suggestion
        yield [false, "<?php\n?><?php\n"];
        yield [false, 'Hello<?php echo "World!"; ?>'];
        yield [false, '<?php echo "Hello"; ?> World!'];
```

File: src/Tokenizer/Tokens.php (hunk 1 above)
Comment (kubawerlos) [reply src/Tokenizer/Tokens.php]: Definitely, I don't know what I was thinking, I've named test case "this-is too-much" and made is passing check 🤦🏼

File: src/Tokenizer/Tokens.php (hunk 2 above)
Comment (kubawerlos) [reply src/Tokenizer/Tokens.php:1103]: Apparently no, and we never had.

File: tests/Tokenizer/TokensTest.php [hunk 6]
```diff
@@ -353,12 +353,24 @@ public function provideMonolithicPhpDetectionCases(): iterable
     {
         yield [true, "<?php\n"];
         yield [true, "<?php\n?>"];
+        yield [false, "#!/usr/bin/bash\ncat <?php\n"];
+        yield [false, "#!/usr/bin/env bash\ncat <?php\n"];
+        yield [true, "#!/usr/bin/php\n<?php\n"];
+        yield [true, "#!/usr/bin/php7.4-cli\n<?php\n"];
```
Comment (keradus) [inline tests/Tokenizer/TokensTest.php:359]:
```suggestion
        yield [true, "#!/usr/bin/php7.4-cli\n<?php\n"];
        yield [false, "#!/usr/bin/php\n\n<?php\n"]; // empty line after shebang would be printed to console before PHP executes
```

File: src/Tokenizer/Tokens.php [hunk 7]
```diff
@@ -1088,17 +1088,19 @@ public function clearRange(int $indexStart, int $indexEnd): void
...
         }
 
-        if ($this->isTokenKindFound(T_INLINE_HTML)) {
+        if ($this->countTokenKind(T_INLINE_HTML) > 1) {
             return false;
         }
 
-        return 1 >= ($this->countTokenKind(T_OPEN_TAG) + $this->countTokenKind(T_OPEN_TAG_WITH_ECHO));
+        if (1 === $this->countTokenKind(T_INLINE_HTML)) {
+            return 1 === Preg::match('/^#!.*$/', $this[0]->getContent());
```
Comment (keradus) [inline src/Tokenizer/Tokens.php:1100]:
```suggestion
            return 1 === Preg::match('/^#!.+$/', $this[0]->getContent());
```

to handle https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/6233/files#r784369829

**Review by keradus — APPROVED** [review]

**Review by SpacePossum — APPROVED** [review]

**Comment by keradus** [comment]
Thank you @kubawerlos.

**Pull request closed without merging**